client/analytics/report/categories/index.js

```diff
@@ -55,6 +62,13 @@ export default class CategoriesReport extends Component {
 				: 'category';
 		}
 
+		const compareProducts = filters[ 0 ].filters.filter(
+			( item ) => item.value === 'compare-categories'
+		);
+		if ( compareProducts.length ) {
```

Comment:
This is much better. The trouble with looking at the length and conditionally adding the `onClick` handler is that it's potentially going to silently do nothing if the filters change shape - it's gone from too brittle to too flexible. I think if you just take out the `if` statement and assume that `compareProducts[0]` isn't going to fail, this would be robust but fail in a way that makes sense and is easier to track down immediately.

Reply:
That's a really good idea 👍 changed it in https://github.com/woocommerce/woocommerce-admin/pull/5649/commits/a1a9552ced71db7a1ad5708892dc30e8bd789653